test/files/neg/t5856.scala

```diff
@@ -8,4 +8,12 @@ object Test {
   val s7 = s"$s1 $null $super"
   val s8 = s"$super"
   val s9 = s"$"
+  val sA = s"$this"
```

Comment:
this is a `neg` test - shouldn't `s"$this"` compile fine? (it appears to in the junit test)

Reply:
this is still useful in a neg test because it tests that no warning / error is issued

Reply:
Sorry Nth, I almost added a comment at that line that the rest compiles. Then I didn't. Maybe I fell asleep? Maybe my spouse arrived home late from work at that moment? This delta was in an old PR that was sadly rejected.

Reply:
oh except I changed assert to assertEquals, just because.